ndcube/ndcube_sequence.py

```diff
@@ -239,6 +245,91 @@ def explode_along_axis(self, axis):
+                table_coords.append(QuantityTableCoordinate(u.Quantity(axis_coords),
+                                                            names=axis_name,
+                                                            physical_types=physical_type))
+
+            elif isinstance(axis_coords[0], Time):
+                table_coords.append(TimeTableCoordinate(Time(axis_coords),
+                                                        names=axis_name,
+                                                        physical_types=physical_type))
+
+            elif isinstance(axis_coords[0], SkyCoordTableCoordinate):
+                table_coords.append(SkyCoordTableCoordinate(SkyCoord(axis_coords),
+                                                            names=axis_name,
+                                                            physical_types=physical_type))
+
+            else:
+                warnings.warn(f"Skipping '{axis_name}'. Unable to convert global_coords "
+                              f"'{axis_name}' into an axis. global_coords must be either "
+                              "`astropy.units.Quantity`, `astropy.time.Time`, or "
+                              "`astropy.coordinates.SkyCoord`.",
+                              UserWarning)
+
+        # If no global coords were found, use the cube index to form an axis.
+        if not table_coords:
+            table_coords.append(QuantityTableCoordinate(u.Quantity(np.arange(len(self.data))),
+                                                        names='sequence',
+                                                        physical_types='meta.obs.sequence'))
+
+        return MultipleTableCoordinate(*table_coords).wcs
+
+    def combine_cubes(self, common_wcs_index=0):
+        """
+        Reprojects all `~ndcube.NDCube` objects to a common WCS, and stacks the data together
+        to return a single (N+1)-dimensional `~ndcube.NDCube` with an associated Compound WCS
+        containing extra axes corresponding to the sequence axis.
+
+        Parameters
+        ----------
+        common_wcs_index: `int`, optional
+            The index of the `~ndcube.NDCube` whose WCS is to be used as the base on which to
+            reproject all other `~ndcube.NDCube`s in the `~ndcube.NDCubeSequence`.
+
+        Returns
+        -------
+        `ndcube.NDCube`
+            A combined (N+1)D cube with stacked data of all cubes.
+        """
+        from ndcube import NDCube  # adding this here to prevent a circular import
+
+        # Verify if all cubes have the same shape.
+        cube_shape = self[0].data.shape
+        for cube in self:
+            if cube.data.shape != cube_shape:
+                raise ValueError('All cubes must have the same shape.')
+
+        # Reproject all cubes to a common WCS
+        target_wcs = self[common_wcs_index].wcs
+        shape_out = self[common_wcs_index].data.shape
+        reprojected_cubes = [cube.reproject_to(target_wcs, shape_out=shape_out,
```

Comment:
maybe a warning or something should be flagged here if this reproject is going to take a long time. Imagine a user has a sequence of a hundred 4Kx4K AIA images and then tried to run this, it will take a looonng time